src/change_stream.ts

```diff
@@ -926,25 +926,30 @@ export class ChangeStreamCursor<
     return this._resumeToken;
   }
 
-  get resumeOptions(): ResumeOptions {
-    const result: ResumeOptions = filterOptions(this.options, CURSOR_OPTIONS);
-
-    if (this.resumeToken || this.startAtOperationTime) {
-      for (const key of ['resumeAfter', 'startAfter', 'startAtOperationTime']) {
-        Reflect.deleteProperty(result, key);
-      }
+  get resumeOptions(): ChangeStreamCursorOptions {
+    const options: ChangeStreamCursorOptions = {
+      ...this.options
+    };
 
-      if (this.resumeToken) {
-        const resumeKey =
-          this.options.startAfter && !this.hasReceived ? 'startAfter' : 'resumeAfter';
+    for (const key of ['resumeAfter', 'startAfter', 'startAtOperationTime'] as const) {
+      delete options[key];
+    }
 
-        result[resumeKey] = this.resumeToken;
-      } else if (this.startAtOperationTime && maxWireVersion(this.server) >= 7) {
-        result.startAtOperationTime = this.startAtOperationTime;
+    if (this.resumeToken != null) {
+      if (this.options.startAfter && !this.hasReceived) {
+        options.startAfter = this.resumeToken;
+      } else {
+        options.resumeAfter = this.resumeToken;
       }
+    } else if (
+      this.resumeToken == null &&
```

Comment:
I can make that change if you want.  My goal here was to port the algorithm as close as possible to the algorithm in the spec, so that it's very easy to compare the two.  Thoughts?

Reply:
I don't think it hurts anything to leave it in, it just doesn't really follow the usual language conventions, so it reads strange, from a js perspective. As far as the algorithm - I think the best way to preserve *that* logic would be to organize our unit tests along the specification that is outlined there (which you pretty much have already done).